src/porepy/models/verification_setups/terzaghi.py

```diff
@@ -0,0 +1,723 @@
+
+    time_manager: pp.TimeManager
+    """Time-stepping object."""
+
+    stress_keyword: str
+    """Key for accessing data parameters for the mechanics subproblem."""
+
+    bc_values_mechanics_key: str
+    """Key for accessing mechanical boundary values."""
+
+    solid: pp.SolidConstants
+    """Subclass of pp.SolidConstants with different default values."""
+
+    fluid: pp.FluidConstants
+    """Subclass of pp.FluidConstants with different default values."""
+
+    results: list[StoreResults]
+    """List of stored result objects."""
+
+    exact_sol: ExactSolution
+    """Exact solution object."""
+
+    # ----> Derived physical quantities
+    def confined_compressibility(self) -> number:
+        """Compute confined compressibility [Pa^-1].
+
+        Returns:
+            Confined compressibility.
+
+        """
+        mu_s = self.solid.shear_modulus()
+        lambda_s = self.solid.lame_lambda()
+        m_v = 1 / (2 * mu_s + lambda_s)
+
+        return m_v
+
+    def consolidation_coefficient(self) -> number:
+        """Compute consolidation coefficient [m^2 * s^-1].
+
+        Returns:
+            Coefficient of consolidation.
+
+        """
+        k = self.solid.permeability()  # [m^2]
+        mu_f = self.fluid.viscosity()  # [Pa * s]
+        rho = self.fluid.density()  # [kg * m^-3]
+        gamma_f = rho * pp.GRAVITY_ACCELERATION  # specific weight [Pa * m^-1]
+        hydraulic_conductivity = (k * gamma_f) / mu_f  # [m * s^-1]
+        storage = self.solid.specific_storage()  # [Pa^-1]
+        if not storage == 0.0:
+            raise ValueError("Terzaghi's solution requires zero specific storage.")
+        alpha_biot = self.solid.biot_coefficient()  # [-]
+        m_v = self.confined_compressibility()  # [Pa^-1]
+
+        c_v = hydraulic_conductivity / (gamma_f * (storage + alpha_biot**2 * m_v))
+
+        return c_v
+
+    # ----> Non-dimensionalization methods
+    def nondim_time(self, t: number) -> float:
+        """Non-dimensionalize time.
+
+        Parameters:
+            t: Time [s].
+
+        Returns:
+            Dimensionless time for the given time `t`.
+
+        """
+        h = self.params["height"]
+        c_v = self.consolidation_coefficient()
+
+        return (t * c_v) / (h**2)
+
+    def nondim_length(
+        self, length: Union[number, np.ndarray]
+    ) -> Union[number, np.ndarray]:
+        """Nondimensionalize length.
+
+        Parameters:
+            length : length in meters.
+
+        Returns:
+            Non-dimensionalized length.
+
+        """
+        return length / self.params["height"]
+
+    def nondim_pressure(self, pressure: np.ndarray) -> np.ndarray:
+        """Nondimensionalize pressure.
+
+        Parameters:
+            pressure : pressure in Pa.
+
+        Returns:
+            Non-dimensional pressure.
+
+        """
+        return pressure / np.abs(self.params["vertical_load"])
+
+    # ----> Postprocessing methods
+    def displacement_trace(
```

Comment:
This one's interesting. Perhaps it should be promoted so that we can use it elsewhere?

Reply:
Of course. Any ideas where?

Reply:
Not quite sure. We could postpone it to the migration of verification setups to applications.

Reply:
Ok. I will add a TODO and resolve the conversation.